astropy/units/physical.py

```diff
@@ -542,9 +544,21 @@ def get_physical_type(obj):
         return PhysicalType(unit, "unknown")
 
 
+# ------------------------------------------------------------------------------
+# Script section creating the physical types and the documentation
+
+# define the physical types
 for unit, physical_type in _units_and_physical_types:
     def_physical_type(unit, physical_type)
 
+# key-view of physical type names
+available_physical_types = _name_physical_mapping.keys()
+
+# set the physical types on the module
+for name, physical_type in _name_physical_mapping.items():
+    name = name.replace(' ', '_').replace('(', '').replace(')', '')
+    setattr(sys.modules[__name__], name, physical_type)
+    __all__.append(name)
```

Comment:
I like the idea of making the physical types available. Note that by appending to `__all__`, they will also be available under the main submodule (i.e., `u.length` will exist), unless we make the import in `astropy/units/__init__.py` more selective.

Reply:
That seems a + ? Os is there a good reason to not flatten the namespace?

Reply:
What I just pushed with __getattr__ and __dir__ removed them from __all__. Should I re-add? I have no strong preference on how flat the namespace should be.

Reply:
The reason that I wonder about flattening the name space is that that is only done once. So, if the user defines a new smorgashbord physical type, `u.smorgashbord` will not work -- unless we also define `__getattr__` and `__dir__` inside `__init__.py`.

My sense would be to split the two concerns - in this PR make `physical.length` etc available and in another think about how to deal with it for the top level.

Reply:
sounds good. removing it now.